x/staking/keeper/liquid_stake_test.go

```diff
@@ -0,0 +1,514 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	simapp "github.com/iqlusioninc/liquidity-staking-module/app"
+	"github.com/iqlusioninc/liquidity-staking-module/x/staking/types"
+	"github.com/stretchr/testify/require"
+)
+
+// Tests Set/Get TotalLiquidStakedTokens
+func TestTotalLiquidStakedTokens(t *testing.T) {
+	_, app, ctx := createTestInput(t)
+
+	// Before it's been initialized, it should return zero
+	require.Equal(t, sdk.ZeroInt(), app.StakingKeeper.GetTotalLiquidStakedTokens(ctx), "zero")
+
+	// Update the total liquid staked
+	total := sdk.NewInt(100)
+	app.StakingKeeper.SetTotalLiquidStakedTokens(ctx, total)
+
+	// Confirm it was updated
+	require.Equal(t, total, app.StakingKeeper.GetTotalLiquidStakedTokens(ctx), "initial")
+}
+
+// Tests Increase/Decrease TotalValidatorTotalLiquidShares
+func TestValidatorTotalLiquidShares(t *testing.T) {
+	_, app, ctx := createTestInput(t)
+
+	// Create a validator address
+	privKey := secp256k1.GenPrivKey()
+	pubKey := privKey.PubKey()
+	valAddress := sdk.ValAddress(pubKey.Address())
+
+	// Set an initial total
+	initial := sdk.NewDec(100)
+	validator := types.Validator{
+		OperatorAddress:   valAddress.String(),
+		TotalLiquidShares: initial,
+	}
+	app.StakingKeeper.SetValidator(ctx, validator)
+}
+
+// Tests AccountIsLiquidStakingProvider
+func TestAccountIsLiquidStakingProvider(t *testing.T) {
+	_, app, ctx := createTestInput(t)
+
+	// Create base account
+	baseAccountAddress := sdk.AccAddress("base-account")
+	app.AccountKeeper.SetAccount(ctx, authtypes.NewBaseAccountWithAddress(baseAccountAddress))
+
+	// Create an ICA module account
+	icaModuleAccountName := "ica-account"
```

Comment:
can we also test a non-ICA module account?

Reply:
that module account name is arbitrary - that's not the actual format of ICA accounts.

The name here is irrelevant for the purposes of the test cause all the `AcountIsLiquidStakingProvider` function does is cast attempt to cast the account and then check the address length